### ISSUE TITLE
nanodbc dependency on libodbc.so.2 not declared in packaging

### ISSUE THREAD

**Issue by trel** [irods/irods#5389]
- [x] master
- [x] 4-2-stable
---------
### iRODS Version, OS and Version

4.2.8, on a catalog consumer

### What did you try to do?

look at clean logfiles

### Expected behavior

clean logfiles

### Observed behavior (including steps to reproduce, if applicable)

```
Feb  2 09:15:49 pid:71953 remote addresses: x.x.x.x ERROR: [-]     /irods_git_repo/lib/core/src/apiHandler.cpp:198:irods::error irods::init_api_table(irods::api_entry_table &, irods::pack_entry_table &, bool) :  status [PLUGIN_ERROR]  errno [] -- message [failed to open shared object file [/usr/lib/irods/plugins/api/libirods_atomic_apply_metadata_operations_server.so] :: dlerror: is [libodbc.so.2: cannot open shared object file: No such file or directory]]
        [-]     /irods_git_repo/lib/core/include/irods_load_plugin.hpp:176:irods::error irods::load_plugin(PluginType *&, const std::string &, const std::string &, const std::string &, const Ts &...) [PluginType = irods::api_entry, Ts = <char [12]>] :  status [PLUGIN_ERROR]  errno [] -- message [failed to open shared object file [/usr/lib/irods/plugins/api/libirods_atomic_apply_metadata_operations_server.so] :: dlerror: is [libodbc.so.2: cannot open shared object file: No such file or directory]]
```

The catalog provider does not have these messages, since the database plugin declares a dependency on libodbc.

Since we have shipped the `libirods_atomic_apply_metadata_operations_server.so`, it depends on nanodbc, and that links against libodbc.

- apt package `libodbc1`
- yum package `unixODBC`

**Comment by SwooshyCueb** [comment]
To clarify, this dependency should be declared in the nanodbc irods-externals package, not in the package that includes `libirods_atomic_apply_metadata_operations_server.so`

**Comment by trel** [comment]
yes, definitely correct.

since we ship `libirods_atomic_apply_metadata_operations_server.so` with the server now, wanted to make sure it was included in 4.2.9.
